src/IssueViz/Editor/LocationTagging/LocationTagger.cs

```diff
@@ -157,21 +159,55 @@ private void SafeOnBufferChange(object sender, TextContentChangedEventArgs e)
                 var newTextSnapshot = e.After;
 
                 TranslateTagSpans(newTextSnapshot);
-                locationService.Refresh(new string[] { FilePath });
+                locationService.Refresh(new[] { FilePath });
 
-                NotifyTagsChanged(newTextSnapshot);
+                var affectedSpan = CalculateSpanOfChanges(e.Changes, newTextSnapshot);
+                NotifyTagsChanged(affectedSpan);
             }
             catch (Exception ex) when (!ErrorHandler.IsCriticalException(ex))
             {
                 logger.WriteLine(Resources.ERR_HandlingBufferChange, ex.Message);
             }
         }
 
-        private void NotifyTagsChanged(ITextSnapshot newTextSnapshot)
+        private void NotifyTagsChanged(SnapshotSpan affectedSpan)
         {
             // Notify the editor that our set of tags has changed
-            var wholeSpan = new SnapshotSpan(newTextSnapshot, 0, newTextSnapshot.Length);
-            TagsChanged?.Invoke(this, new SnapshotSpanEventArgs(wholeSpan));
+            TagsChanged?.Invoke(this, new SnapshotSpanEventArgs(affectedSpan));
+        }
+
+        /// <summary>
+        /// Method calculates the span from the start of (old+new) TagSpans and until the end of (old+new) TagSpans
+        /// </summary>
+        private SnapshotSpan CalculateSpanOfAllTags(IList<ITagSpan<IIssueLocationTag>> oldTags, ITextSnapshot textSnapshot)
+        {
+            var allTagSpans = TagSpans
+                .Union(oldTags ?? Array.Empty<ITagSpan<IIssueLocationTag>>())
+                .Select(x => x.Span.Span)
+                .ToArray();
+
+            return CalculateAffectedSpan(textSnapshot, allTagSpans, allTagSpans);
+        }
+
+        /// <summary>
+        /// Method calculates the span from the start of the editor changes and until the end of TagSpans
+        /// </summary>
+        private SnapshotSpan CalculateSpanOfChanges(INormalizedTextChangeCollection changeCollection, ITextSnapshot newTextSnapshot)
+        {
+            var startSpans = changeCollection.Select(x => x.NewSpan).ToArray();
+            var endSpans = TagSpans.Select(x => x.Span.Span);
+            
+            return CalculateAffectedSpan(newTextSnapshot, startSpans, endSpans);
+        }
+
+        private static SnapshotSpan CalculateAffectedSpan(ITextSnapshot textSnapshot, IEnumerable<Span> startSpans, IEnumerable<Span> endSpans)
```

Comment:
In the TBIT [we used to call Span.TranslateTo](https://github.com/SonarSource/sonarlint-visualstudio/blob/master/src/Integration.Vsix/SonarLintTagger/TextBufferIssueTracker.cs#L215) with PointTrackingMode.Negative / PointTrackingMode.Positive and I don't do that. I'm not sure if I need to, the editor seems to behave ok without it.

Reply:
All of our other taggers still tell the editor to refresh everything so I think the editor would still look ok even if this code isn't correct.
Let's discuss this offline. I think the translation is necessary when the set of issues changes. When the buffer changes, it might be enough to just use the changed span.